src/ir/typescript_types.py

```diff
@@ -216,6 +232,86 @@ def get_name(self):
         return 'undefined'
 
 
+class NumberLiteralType(TypeScriptBuiltin):
+    def __init__(self, literal, name="NumberLiteralType", primitive=False):
+        super().__init__(name, primitive)
+        self.literal = literal
+        self.supertypes.extend([ObjectType(), NumberType()])
+
+    def get_literal(self):
+        return self.literal
+
+    def is_assignable(self, other):
```

Comment:
Could you please add a comment about when a type is a assignable to a num literal type?

Reply:
Refer to my comment above.